test_suite/cloud/test_aws.py

```diff
@@ -558,12 +558,25 @@ def test_yum_group_install(self, host):
             pytest.skip('Not applicable to Atomic host AMIs')
 
         with host.sudo():
-            assert host.run_test('yum -y groupinstall "Development tools"'), \
-                'Error while installing Development tools group'
-
-            package_to_check = 'glibc-devel'
-            assert host.package(package_to_check).is_installed, \
-                f'{package_to_check} is not installed'
+            attempts = 2
+            for attempt in range(attempts):
```

Comment:
Proposal to reduce nesting and loops:

```
dev_tools_install_command = 'yum -y groupinstall "Development tools"'

result = test_lib.print_host_command_output(host, dev_tools_install_command, capture_result=True)

if result.failed:
    err_message = "This system is not registered to Red Hat Subscription Management"
    if err_message in result.stderr:
        print('"Development tools" installation attempt failed. Trying to apply a workaround...')

        host.run('echo -e "enabled=0" > /etc/yum/pluginconf.d/subscription-manager.conf')
        host.run('yum clean all')

result_second_attempt = test_lib.print_host_command_output(host, dev_tools_install_command, capture_result=True)

assert result_second_attempt.succeeded, \
    'Error while installing Development tools group after two attempts. Check test case output for more details.'

print('"Development tools" installed successfully.')

package_to_check = 'glibc-devel'
assert host.package(package_to_check).is_installed, \
    f'{package_to_check} is not installed'

```

* `print_host_command_output`: To run the command in the host, and at the same time pretty-print the output (either if it's an error it will be pretty-printed in the test case output). By using `capture_result=True`, we make sure to retrieve the `CommandResult` object for further checks later in the code.
* Some additional print lines to inform us in the test case what happened and which steps took place during test runtime (for further debugging if needed).

Reply:
How should we handle any other type of errors? I remember this test case was a "victim" of the DB_PAGE_NOT_FOUND issue when rpm database is broken after boot, for some reason.
In that case, should we do a second attempt? Or fail inmediately?

Reply:
I agree, seems like a better approach!
Thanks

Reply:
Regarding the prints, maybe we should use instead testing logs for the steps we do, I see that "print_divider" is something that is being used in CIV, what do you think?

Reply:
Although the "print_divider" doesn't have the options to set logging levels like ( INFO/ ERRORs etc...)
I think it's a better practice for debugging

Reply:
Oh yes I forgot about that one! That's great for delimiting steps :+1:

Reply:
I think I will create a separate PR to set the logging because the 'print devider' doesn't show the output in the console, which might also help when debugging just from Jenkins console output or locally.
There's a logging module that exist in Python, it's pretty common and it does the job: prints the output to console during run time and also in the final reports.

Reply:
We have a ticket in our backlog to implement better logging the whole project.
For now let's keep it simple and the current solution does the job as you said. :+1: